queue/ring.go

```diff
@@ -128,9 +129,24 @@ L:
 // to the queue or Dispose is called on the queue.  An error will be returned
 // if the queue is disposed.
 func (rb *RingBuffer) Get() (interface{}, error) {
-	var n *node
-	pos := atomic.LoadUint64(&rb.dequeue)
-	i := 0
+	return rb.Poll(0)
+}
+
+// Poll will return the next item in the queue.  This call will block
+// if the queue is empty.  This call will unblock when an item is added
+// to the queue, Dispose is called on the queue, or the timeout is reached. An
+// error will be returned if the queue is disposed or a timeout occurs. A
+// non-positive timeout will block indefinitely.
+func (rb *RingBuffer) Poll(timeout time.Duration) (interface{}, error) {
```

Comment:
Yeah, I wrote a little app that allocated half a gig of byte arrays and then used some goroutines to grab items out of the pool and then dump them back in.  I went and looked at the pool source code and it's going to be tough to beat:
https://github.com/golang/go/blob/master/src/sync/pool.go

They have something that looks akin to threadlocal storage that caches an item so the thread can quickly grab another.  Going to be really difficult to match that performance with any sort of FIFO queue.  Cheaters.